FlagsmithClient/Classes/Flagsmith.swift

```diff
@@ -55,15 +73,34 @@ public class Flagsmith {
     if let identity = identity {
       getIdentity(identity) { (result) in
         switch result {
-        case .success(let identity):
-          completion(.success(identity.flags))
+        case .success(let thisIdentity):
+          self.updateCache(flags: thisIdentity.flags, forIdentity: identity)
+          completion(.success(self.getFlagsUsingCacheAndDefaults(flags: thisIdentity.flags, forIdentity: identity)))
         case .failure(let error):
-          completion(.failure(error))
+          let fallbackFlags = self.getFlagsUsingCacheAndDefaults(flags: [], forIdentity: identity)
+          if fallbackFlags.isEmpty {
+            completion(.failure(error))
+          }
+          else {
+            completion(.success(fallbackFlags))
+          }
         }
       }
     } else {
       apiManager.request(.getFlags) { (result: Result<[Flag], Error>) in
-        completion(result)
+        switch result {
+        case .success(let flags):
+          self.updateCache(flags: flags, forIdentity: identity)
```

Comment:
True, it is always nil in the else block, but really we're just saying pass whatever identity what passed into this method, whether it is nil or not. I think that's safer since if for example someone refactored it and moved around the else block, they may miss the identity part and not pass it through.

Reply:
I guess this is why it would be nice to have unit test coverage here, but that's another conversation! 

For readability's sake, I do think that I prefer to be explicit here, so I think we should either omit the `forIdentity` parameter (as I think it defaults to `nil`) or pass `nil` explicitly here. That being said, I'm not sure that my feelings on this one are strong enough to argue if you are keen to leave it as it is.